WordPress/openverse
# Use the `VMediaCollection` for search and collection results

## Fixes


Based on #3831
Fixes #3884 by @obulat
Fixes #3576 by @AetherUnbound
Fixes #3830 by @obulat

_Initially, I split the changes into the search result and collection result changes. However, this made it more difficult to understand the scope and changes, and to test the PRs. This is why I merged the collection changes (as well as the deletion of the now-unused components) into this PR._

## Description


This PR fixes the collection result fetching and improves the fetching of the search results. 


It uses the new `VMediaCollection` component and 2 new components for the search and collection results.
- `VSearchResults` component fills in the components necessary for search results: search title and content links in the header, load more button, and the external search form in the footer.
- `VCollectionResults` component adds the components necessary for collection results: `VCollectionHeader` and load more button.

The media fetching was updated: now it is done in a single place: `search.vue` and `collection.vue`, instead of being split among these pages, middleware and the Load more component. This allows us to handle fetching, refetching, and loading more items from the same place. The "Load more" button now only handles sending the Analytics events.

## Testing Instructions

Run the app with `additional_search_views` on.
Check that the search results are updated correctly: when the search term is updated, search type is changed, filters are updated or load more button is clicked. The result labels should be updated correctly, loading skeletons shown when media is fetching for the first time, media should be updated for the correct query.
The same check needs to be done for collection pages

The search results should be updated correctly when the `additional_search_views` is off.

Repeat the instructions from #3884 and see that the problem is fixed and results are refreshed

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
